source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/dialogs/connect/ServerAuthDialog.java

```diff
@@ -0,0 +1,161 @@
+
+        final Composite buttonComposite = new Composite(dialogArea, SWT.NONE);
+        final GridLayout buttonCompositeLayout = new GridLayout(2, false);
+        buttonCompositeLayout.horizontalSpacing = getHorizontalSpacing();
+        buttonCompositeLayout.verticalSpacing = getVerticalSpacing();
+        buttonCompositeLayout.marginWidth = 0;
+        buttonCompositeLayout.marginHeight = 0;
+        buttonComposite.setLayout(buttonCompositeLayout);
+
+        userpasswordButton = new Button(buttonComposite, SWT.RADIO);
+        userpasswordButton.setText(Messages.getString("ServerAuthDialog.UserPasswordTypeLabel")); //$NON-NLS-1$
+        userpasswordButton.setSelection(true);
+        // userpasswordButton.addSelectionListener(new SelectionAdapter() {
+        // @Override
+        // public void widgetSelected(final SelectionEvent e) {
+        // onCredentialTypeSelected();
+        // }
+        // });
+
+        patButton = new Button(buttonComposite, SWT.RADIO);
+        patButton.setText(Messages.getString("ServerAuthDialog.PatTypeLabel")); //$NON-NLS-1$
+        patButton.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(final SelectionEvent e) {
+                onCredentialTypeSelected();
+            }
+        });
+
+        final Label usernameLabel = new Label(dialogArea, SWT.NONE);
+        usernameLabel.setText(Messages.getString("ServerAuthDialog.UsernameLabel")); //$NON-NLS-1$
+
+        usernameText = new Text(dialogArea, SWT.BORDER);
+        GridDataBuilder.newInstance().hGrab().hFill().applyTo(usernameText);
+
+        final Label passwordLabel = new Label(dialogArea, SWT.NONE);
+        passwordLabel.setText(Messages.getString("ServerAuthDialog.PasswordTokenLabel")); //$NON-NLS-1$
+
+        passwordText = new Text(dialogArea, SWT.PASSWORD | SWT.BORDER);
+        GridDataBuilder.newInstance().hGrab().hFill().wCHint(passwordText, 16).applyTo(passwordText);
+
+        if (credentials == null || credentials instanceof PatCredentials) {
+            patButton.setSelection(true);
+            userpasswordButton.setSelection(false);
+        } else if (credentials instanceof UsernamePasswordCredentials) {
+            savedUserName = ((UsernamePasswordCredentials) credentials).getUsername();
+            usernameText.setText(savedUserName);
+            patButton.setSelection(false);
+            userpasswordButton.setSelection(true);
+        }
+
+        if (usernameText.getText().length() > 0) {
+            passwordText.setFocus();
+        }
+
+        onCredentialTypeSelected();
+    }
+
+    private void onCredentialTypeSelected() {
+        if (patButton.getSelection()) {
+            savedUserName = usernameText.getText();
+            usernameText.setText(StringUtil.EMPTY);
+        } else {
+            usernameText.setText(savedUserName);
+        }
+        usernameText.setEnabled(userpasswordButton.getSelection());
+    }
```

Comment:
Some users may be confused if they can enter a username with a personal access token.  If they select PAT as the credential type, can we keep the username field disabled or hide it and give it some default value?  Maybe you're already doing something like this and I just need to see the UI.